### ISSUE TITLE
error in NIFTItoNRRD handling of space directions

### ISSUE THREAD

**Issue by vmagnotta** [BRAINSia/BRAINSTools#310]
There seems to be an issue with the proper handling of image orientation in DWIConvert. If a DICOM series is directly converted to NiFTI versus the conversion from DICOM->NRRD->NIFTI the NIFTI sform matrix is not consistent while the qform matrix is. Slicer shows the image in the same orientation likely because it is using the qform orientation. 

The problem is related to the srow_z rotation portion of the transform which is getting multiplied by -1.  The translation part of the matrix is correct.

**Comment by hjmjohnson** [comment]
@vmagnotta Vince,  I'm looking at this code now.    Do you happen to know where in the code the problem is occurring?

**Comment by vmagnotta** [comment]
Hans,

I looked at this some on Friday. Since both transforms originate from the image direction matrix, it would assume the bug could be in the helper function that generates the sform matrix from the direction cosines.

Vince


--
Associate Professor
Department of Radiology
L311 PBDB
169 Newton Road
Iowa City, IA 52242
E-mail: vincent-magnotta@uiowa.edu
Phone: 319-335-5482 Fax: 319-353-6275
Website: http://www.healthcare.uiowa.edu/Radiology

**Comment by hjmjohnson** [comment]
Thanks Vince.  I’ll need to look into this again.  I thought that if the qform is set, then the sform is NOT set, and is not used.  

 

Hans

 

 

--

**Comment by vmagnotta** [comment]
Hans

That may be the case. It is odd that the sform is wrong while the qform is correct.

Vince

---
Associate Professor
Department of Radiology
L311 PBDB
169 Newton Road
Iowa City, IA 52242
E-mail: vincent-magnotta@uiowa.edu<mailto:vincent-magnotta@uiowa.edu>
Phone: 319-335-5482 Fax: 319-353-6275
Website: http://www.healthcare.uiowa.edu/Radiology

On Nov 19, 2016, at 8:07 PM, Hans Johnson <notifications@github.com<mailto:notifications@github.com>> wrote:

Thanks Vince.  I’ll need to look into this again.  I thought that if the qform is set, then the sform is NOT set, and is not used.



Hans





--

**Comment by hjmjohnson** [comment]
Vince,

 

I am working through several issues with properly representing Nrrd files and Nifti files from DWIConvert.

 

 

Another popular tool for many dwi processing to convert dicom to NIFTI is the dcm2niix tool.   I’ve previously had conversations with the author of that tool, and I thought we had resolved the interpretation issues.

 

https://github.com/rordenlab/dcm2niix/issues/40

 

This is a discussion about some data that I am working with that is from BC Ho’s study.

 

Could you please comment if you think the comments are correct, or if you think that perhaps there is a mis-interpretation of the data that I sent?

 

Thanks,

Hans

 

 

--

**Comment by vmagnotta** [comment]
Hans,

There are two slice orderings that occur with Mosaic images from Siemens. These are completely independent.


1.       There is the slice acquisition order. This determines the order of slices and how they are acquired temporally.

2.       The second is the image sorting. This determines the order that the images are arranged in the Mosaic image. This independent of the acquisition order. This is done for convenience to sort the images in the preferred order for the radiologists.

Chris may have never changed the image sorting order of the image and that is why he has not seen the change in slice ordering within a Mosaic image.

Vince


--
Associate Professor
Department of Radiology
L311 PBDB
169 Newton Road
Iowa City, IA 52242
E-mail: vincent-magnotta@uiowa.edu
Phone: 319-335-5482 Fax: 319-353-6275
Website: http://www.healthcare.uiowa.edu/Radiology

**Comment by hjmjohnson** [comment]
Vince,

 

That is helpful information.  Do you happen to know where to find the DICOM tags that can be used to interpret those conditions?

 

Thanks,
Hans

 

 

--

**Comment by vmagnotta** [comment]
I know that this must be stored in the Siemens Private header fields of either 0029, 1010 or 0029, 1020. I will try to dig up more information of this.

Vince

--
Associate Professor
Department of Radiology
L311 PBDB
169 Newton Road
Iowa City, IA 52242
E-mail: vincent-magnotta@uiowa.edu<mailto:vincent-magnotta@uiowa.edu>
Phone: 319-335-5482 Fax: 319-353-6275
Website: http://www.healthcare.uiowa.edu/Radiology

**Comment by hjmjohnson** [comment]
Thank you Vince.

 

Hans

 

 

--

**Comment by hjmjohnson** [comment]
@vmagnotta I think this is now fixed in #315 

I have a test suite of 85 visualizations using both fslview and Slicer to ensure that corrections are done.  I'm through 15 visualization validations (a problematic phillips, a GE, and a Siemens), and will complete the rest of the data visualizations tomorrow.

dcm2niix    -> view in fslview
DWIConvert:DicomToNrrd -> View with Slicer
DWIConvert:DicomToFSL -> view in fslview
dcm2niix, DWIConvert:FSLToNrrd -> View in Slicer
DWIConvert:DicomToFSL, DWIConvert:FSLToNrrd ->View in Slicer

[VisualizationValidation.pdf](https://github.com/BRAINSia/BRAINSTools/files/644847/VisualizationValidation.pdf)


I also added a new feature described in #314 , and updated the version number to 4.8.0.
